src/UI/Component/Dropzone/File/File.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace ILIAS\UI\Component\Dropzone\File;
+
+use ILIAS\Data\DataSize;
+use ILIAS\UI\Component\Component;
+use ILIAS\UI\Component\Droppable;
+
+/**
+ * Interface File
+ *
+ * A dropzone where one can drop files on it to be uploaded on the server.
+ *
+ * @author  nmaerchy <nm@studer-raimann.ch>
+ *
+ * @package ILIAS\UI\Component\Dropzone\File
+ */
+interface File extends Component, Droppable {
+
+	/**
+	 * Get a dropzone like this where the files are uploaded to the given URL.
+	 *
+	 * @param string $url
+	 * @return $this
+	 */
+	public function withUploadUrl($url);
+
+	/**
+	 * Get the upload URL where the files are uploaded.
+	 *
+	 * @return string
+	 */
+	public function getUploadUrl();
+
+	/**
+	 * Get a dropzone like this only accepting the submitted file types for uploading, e.g.
+	 * ['jpg', 'png', 'gif'] to allow some image formats.
+	 *
+	 * @param array $types
+	 * @return $this
+	 */
+	public function withAllowedFileTypes(array $types);
+
+	/**
+	 * @return array
+	 */
+	public function getAllowedFileTypes();
+
+	/**
+	 * Get a dropzone like this, restricting the max number of files that can be uploaded.
+	 *
+	 * @param int $max
+	 * @return $this
+	 */
+	public function withMaxFiles($max);
+
+	/**
+	 * Get the max number of files that can be uploaded.
+	 *
+	 * @return int
+	 */
+	public function getMaxFiles();
+
+	/**
+	 * Get a dropzone like this, restricting the max file size of the files to the given limit.
+	 *
+	 * @param DataSize $limit
+	 * @return $this
+	 */
+	public function withFileSizeLimit(DataSize $limit);
+
+	/**
+	 * Get the max file size.
+	 *
+	 * @return DataSize
+	 */
+	public function getFileSizeLimit();
+
+	/**
+	 * Get a dropzone like this, allowing to set the filename for each file being uploaded.
+	 * The custom file name is sent as POST parameter along with the uploaded file.
+	 *
+	 * @param bool $state True to enable custom file names
+	 * @return $this
+	 */
+	public function withCustomFileNames($state);
+
+	/**
+	 * Check if the dropzone supports to enter custom file names for each file being uploaded.
+	 *
+	 * @return bool
+	 */
+	public function allowsCustomFileNames();
+
+	/**
+	 * Get a dropzone like this, allowing to set a description for each file being uploaded.
+	 * The description is sent as POST parameter along with the uploaded file.
+	 *
+	 * @param bool $state True to enable file descriptions
+	 * @return $this
+	 */
+	public function withFileDescriptions($state);
```

Comment:
I had the same understanding problem then with `withCustomFileNames` here, when reading I thought this would already set some description. Maybe introduce some "user-defined" wording here: `withUserDefinedFileNamesEnabled` and `withUserDefinedDescriptionEnabled`?